<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">

<head>

  <meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.56">

  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

  <meta name="author" content="Albert C. Halim, Archer Liu, Stephanie Wu, &amp; Ziyuan Zhao">
  <meta name="dcterms.date" content="2024-11-29">

  <title>Predicting Heart Disease from Cleveland Database</title>
  <style>
    code {
      white-space: pre-wrap;
    }

    span.smallcaps {
      font-variant: small-caps;
    }

    div.columns {
      display: flex;
      gap: min(4vw, 1.5em);
    }

    div.column {
      flex: auto;
      overflow-x: auto;
    }

    div.hanging-indent {
      margin-left: 1.5em;
      text-indent: -1.5em;
    }

    ul.task-list {
      list-style: none;
    }

    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em;
      /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */
      vertical-align: middle;
    }

    /* CSS for citations */
    div.csl-bib-body {}

    div.csl-entry {
      clear: both;
      margin-bottom: 0em;
    }

    .hanging-indent div.csl-entry {
      margin-left: 2em;
      text-indent: -2em;
    }

    div.csl-left-margin {
      min-width: 2em;
      float: left;
    }

    div.csl-right-inline {
      margin-left: 2em;
      padding-left: 1em;
    }

    div.csl-indent {
      margin-left: 2em;
    }
  </style>


  <script src="site_libs/quarto-nav/quarto-nav.js"></script>
  <script src="site_libs/quarto-nav/headroom.min.js"></script>
  <script src="site_libs/clipboard/clipboard.min.js"></script>
  <script src="site_libs/quarto-search/autocomplete.umd.js"></script>
  <script src="site_libs/quarto-search/fuse.min.js"></script>
  <script src="site_libs/quarto-search/quarto-search.js"></script>
  <meta name="quarto:offset" content="./">
  <script src="site_libs/quarto-html/quarto.js"></script>
  <script src="site_libs/quarto-html/popper.min.js"></script>
  <script src="site_libs/quarto-html/tippy.umd.min.js"></script>
  <script src="site_libs/quarto-html/anchor.min.js"></script>
  <link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
  <link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet"
    id="quarto-text-highlighting-styles">
  <script src="site_libs/bootstrap/bootstrap.min.js"></script>
  <link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
  <link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
  <script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


  <link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

  <div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
        <div class="navbar-brand-container mx-auto">
          <a class="navbar-brand" href="./index.html">
            <span class="navbar-title">Heart Disease Predictor</span>
          </a>
        </div>
        <div id="quarto-search" class="" title="Search"></div>
        <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse"
          aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation"
          onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarCollapse">
          <ul class="navbar-nav navbar-nav-scroll me-auto">
            <li class="nav-item">
              <a class="nav-link active" href="./index.html" aria-current="page">
                <span class="menu-text">Home</span></a>
            </li>
            <li class="nav-item">
              <span class="nav-link">
                <span class="menu-text"></span>
              </span>
            </li>
          </ul>
        </div> <!-- /navcollapse -->
        <div class="quarto-navbar-tools">
        </div>
      </div> <!-- /container-fluid -->
    </nav>
  </header>
  <!-- content -->
  <div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
    <!-- sidebar -->
    <!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
      <nav id="TOC" role="doc-toc" class="toc-active">
        <h2 id="toc-title">Table of Contents</h2>

        <ul class="collapse">
          <li><a href="#summary" id="toc-summary" class="nav-link active" data-scroll-target="#summary"><span
                class="header-section-number">1</span> Summary</a></li>
          <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span
                class="header-section-number">2</span> Introduction</a></li>
          <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods"><span
                class="header-section-number">3</span> Methods</a>
            <ul class="collapse">
              <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span
                    class="header-section-number">3.1</span> Data</a></li>
              <li><a href="#preprocessing" id="toc-preprocessing" class="nav-link"
                  data-scroll-target="#preprocessing"><span class="header-section-number">3.2</span> Preprocessing</a>
              </li>
              <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis"><span
                    class="header-section-number">3.3</span> Analysis</a></li>
            </ul>
          </li>
          <li><a href="#results-discussion" id="toc-results-discussion" class="nav-link"
              data-scroll-target="#results-discussion"><span class="header-section-number">4</span> Results &amp;
              Discussion</a>
            <ul class="collapse">
              <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span
                    class="header-section-number">4.1</span> Discussion</a></li>
            </ul>
          </li>
          <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span
                class="header-section-number">5</span> References</a></li>
        </ul>
        <div class="quarto-alternate-formats">
          <h2>Other Formats</h2>
          <ul>
            <li><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li>
          </ul>
        </div>
      </nav>
    </div>
    <!-- main -->
    <main class="content" id="quarto-document-content">

      <header id="title-block-header" class="quarto-title-block default">
        <div class="quarto-title">
          <h1 class="title">Predicting Heart Disease from Cleveland Database</h1>
        </div>



        <div class="quarto-title-meta">

          <div>
            <div class="quarto-title-meta-heading">Author</div>
            <div class="quarto-title-meta-contents">
              <p>Albert C. Halim, Archer Liu, Stephanie Wu, &amp; Ziyuan Zhao </p>
            </div>
          </div>

          <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">November 29, 2024</p>
            </div>
          </div>


        </div>



      </header>


      <section id="summary" class="level1" data-number="1">
        <h1 data-number="1"><span class="header-section-number">1</span> Summary</h1>
        <p>In this project, we developed and evaluated several classification models to predict the presence of heart
          disease using the Cleveland Heart Disease dataset <span class="citation" data-cites="Detrano1988">(<a
              href="#ref-Detrano1988" role="doc-biblioref">Detrano et al. 1988</a>)</span>, which includes various
          clinical features. We compared four models: Logistic Regression, Support Vector Classifier (SVC), Dummy
          Classifier (as a baseline), and Decision Tree Classifier. Logistic Regression performed the best, achieving
          high accuracy of 0.84 and providing interpretable coefficients that helped us understand the impact of each
          feature on heart disease prediction. The SVC also performed well but slightly lagged behind Logistic
          Regression in test accuracy with 0.82. The Dummy Classifier served as a baseline, emphasizing the need for
          more sophisticated models, while the Decision Tree Classifier showed reasonable performance but tended to
          overfit. Misclassifications were analyzed to identify potential feature engineering opportunities, and future
          work could include exploring alternative classifiers such as Random Forests. Additionally, incorporating
          probability estimates into predictions would enhance the modelâ€™s clinical usability, providing clinicians with
          more confidence in the results.</p>
      </section>
      <section id="introduction" class="level1" data-number="2">
        <h1 data-number="2"><span class="header-section-number">2</span> Introduction</h1>
        <p>According to a 2022 CDC federal report, the leading cause of death in the United States is heart disease
          <span class="citation" data-cites="CDC2022">(<a href="#ref-CDC2022" role="doc-biblioref">Centers for Disease
              Control and Prevention (CDC) 2022</a>)</span>. At 702,880 deaths that year, the disease claimed the most
          lives out of popular longevity doctor Peter Attiaâ€™s so-called four horsemen, and nearly as many as cancer and
          diabetes (two of his other horsemen) combined <span class="citation" data-cites="Attia2023">(<a
              href="#ref-Attia2023" role="doc-biblioref">Attia 2023</a>)</span>. Despite past efforts having led to
          promising declines in mortality rates, the disease has gained traction within the last 5 years in
          particular<span class="citation" data-cites="Bui2024">(<a href="#ref-Bui2024" role="doc-biblioref">Bui
              2024</a>)</span>. As such, early detection of heart disease, not to mention increased understanding of and
          heightened mindfulness around mitigating the risk factors for heart disease, can help improve countless lives
          in the United States and elsewhere.
        </p>
        <p>Here we ask if we can use a machine learning algorithm to predict whether an individual has the presence of
          heart disease given a relevant selection of their bioinformatic data. Answering this question is important
          because both patients and their health teams can seek to benefit from tooling and technologies that help in
          the diagnostic process of such a prevalent disease. Given such prevalence, not to mention the potential
          gravity of heart-related conditions, detecting heart disease accurately and early on with a scalable solution
          in medical settings can help enhance medical care in terms of both timeliness and preparedness, to name a few
          aspects. Thus, if a machine learning algorithm can accurately and effectively predict whether an individual
          may harbour this disease early on, this could advance the timeline of early intervention, scale heart disease
          diagnosis efforts, and lead to better patient outcomes, as well as reduce the risk of future complications
          implicated with having heart disease.</p>
      </section>
      <section id="methods" class="level1" data-number="3">
        <h1 data-number="3"><span class="header-section-number">3</span> Methods</h1>
        <section id="data" class="level2" data-number="3.1">
          <h2 data-number="3.1" class="anchored" data-anchor-id="data"><span class="header-section-number">3.1</span>
            Data</h2>
          <p>For this project, we will be using the Heart Disease UCI dataset created by R. Detrano, A. JÃ¡nosi, W.
            Steinbrunn, M. Pfisterer, J. Schmid, S. Sandhu, K. Guppy, S. Lee, and V. Froelicher at the Department of
            Medicine, Veterans Administration Medical Center, Long Beach, California <span class="citation"
              data-cites="Detrano1988">(<a href="#ref-Detrano1988" role="doc-biblioref">Detrano et al. 1988</a>)</span>.
            It was sourced from the UC Irvine Machine Learning Repository <span class="citation"
              data-cites="Detrano1988">(<a href="#ref-Detrano1988" role="doc-biblioref">Detrano et al. 1988</a>)</span>
            and can be found <a href="https://archive.ics.uci.edu/dataset/45/heart+disease">here</a>. The specific file
            used represents the Cleveland locality. The dataset contains 303 rows, with each row representing summary
            statistics for a particular patient, and 14 columns with 13 features and 1 target variable. The target
            variable is the diagnosis of heart disease (angiographic disease status), and the value 0 is for no
            diagnosis of heart disease and the value 1 is for the diagnosis of heart disease. The 13 features are as
            follows:</p>
          <ul>
            <li>Age</li>
            <li>Sex</li>
            <li>Chest pain type</li>
            <li>Resting blood pressure</li>
            <li>Serum cholesterol</li>
            <li>Fasting blood sugar</li>
            <li>Resting electrocardiographic</li>
            <li>Maximum heart rate achieved</li>
            <li>Exercise induced angina</li>
            <li>Oldpeak = ST depression induced by exercise relative to rest</li>
            <li>The slope of the peak exercise ST segment</li>
            <li>Number of major vessels</li>
            <li>Thalassemia blood disorder</li>
          </ul>
          <p>They are encoded in the dataset as follows:</p>
          <ul>
            <li>#3 (age)</li>
            <li>#4 (sex)</li>
            <li>#9 (cp)</li>
            <li>#10 (trestbps)</li>
            <li>#12 (chol)</li>
            <li>#16 (fbs)</li>
            <li>#19 (restecg)</li>
            <li>#32 (thalach)</li>
            <li>#38 (exang)</li>
            <li>#40 (oldpeak)</li>
            <li>#41 (slope)</li>
            <li>#44 (ca)</li>
            <li>#51 (thal)</li>
            <li>#58 (num) (the predicted attribute)</li>
          </ul>
        </section>
        <section id="preprocessing" class="level2" data-number="3.2">
          <h2 data-number="3.2" class="anchored" data-anchor-id="preprocessing"><span
              class="header-section-number">3.2</span> Preprocessing</h2>
          <p>The preprocessing steps are crucial for preparing the heart disease dataset for analysis and model
            training. This section outlines the steps taken to clean and transform the raw data before it is split into
            training and test datasets.</p>
          <p>Firstly, directories are created to store raw and processed data. If the directories do not already exist,
            they are created programmatically.</p>
          <p>Next, the heart disease dataset is downloaded as a ZIP file from the UCI Machine Learning Repository. Once
            downloaded, the ZIP file is extracted to the raw directory.</p>
          <p>After extraction, the dataset is read into a Pandas DataFrame. The column names are defined manually, as
            the dataset does not include a header row. At this point, it was noted that the raw dataset uses the string
            â€˜?â€™ to represent missing values. These values are replaced with NaN to facilitate proper handling during
            further analysis.</p>
          <p>As briefly mentioned in the above section, the target variable, <code>num</code>, represents the presence
            or absence of heart disease. The original values of num range from 0 to 4, where any value greater than 1
            indicates the presence of heart disease. To simplify the analysis, all values greater than 1 are mapped to 1
            (i.e., indicating the presence of heart disease).</p>
          <p>Finally, to prepare the dataset for model training and evaluation, it is split into training and test
            subsets. The split is done in a 70/30 ratio, with stratification to ensure that the distribution of the
            target variable num is preserved in both subsets. The resulting training and test sets are then saved as CSV
            files for future use such as in the next steps, which will involve analyzing the data, training models, and
            assessing their performance.</p>
        </section>
        <section id="analysis" class="level2" data-number="3.3">
          <h2 data-number="3.3" class="anchored" data-anchor-id="analysis"><span
              class="header-section-number">3.3</span> Analysis</h2>
          <p>In this project, we used the Logistic Regression, SVC, Decision Tree, and Dummy Classifier as a baseline to
            build a classification model aimed at predicting the presence of heart disease based on clinical features.
            We used all available features from the dataset, excluding some variables related to the error of certain
            measurements. The data was split into a training set (70%) and a test set (30%). To choose the best value
            for the hyperparameter k, we used 5-fold cross-validation, with accuracy as the classification metric. We
            also standardized the data before fitting the model to ensure the features were on a similar scale. The
            analysis was carried out using Python, with the following libraries: NumPy, Pandas, scikit-learn, and
            Matplotlib.</p>
        </section>
      </section>
      <section id="results-discussion" class="level1" data-number="4">
        <h1 data-number="4"><span class="header-section-number">4</span> Results &amp; Discussion</h1>
        <div style="background-color:#cce7ff; padding:15px; border-radius:5px; border: 2px solid #007acc;">
          <h2 style="color:#007acc;" class="anchored" data-anchor-id="results-discussion">ðŸ“Š Visualization Section</h2>
          <p style="font-size:14px; color:#005f99;">
            The following plots provide insights into the dataset, including target variable distribution, categorical
            feature relationships,
            and model performance (confusion matrix). Each visualization highlights critical aspects of the analysis.
          </p>
        </div>
        <section id="data-set-summary" class="level3" data-number="4.0.1">
          <h3 data-number="4.0.1" class="anchored"><span class="header-section-number">4.0.1</span> Data Set Summary
          </h3>
          <p>The heart disease dataset used in this project is obtained from the UC Irvine Machine Learning Repository.
            The dataset contains 13 features, and the target is a binary variable (<code>num</code>) where:</p>
          <ul>
            <li><strong><code>0</code></strong>: No presence of heart disease</li>
            <li><strong><code>1</code> or higher</strong>: Presence of heart disease.</li>
          </ul>
          <p>Out of the 13 features: - <strong>8 are categorical</strong> (e.g., <code>sex</code>, <code>cp</code>,
            <code>thal</code>). - <strong>5 are numeric</strong> (e.g., <code>age</code>, <code>chol</code>,
            <code>thalach</code>).
          </p>
          <p>These features include various physiological parameters, such as: - <strong>Resting blood
              pressure</strong>, - <strong>Serum cholesterol levels</strong>, - <strong>Maximum heart rate
              achieved</strong>.</p>
          <p>Additionally, it records potential signs of heart disease, such as chest pain type (<code>cp</code>) and
            exercise-induced angina (<code>exang</code>).</p>
          <p>The dataset contains <strong>303 observations</strong>, and the original study used a Bayesian model to
            estimate the probability of having heart disease <span class="citation" data-cites="Detrano1988">(<a
                href="#ref-Detrano1988" role="doc-biblioref">Detrano et al. 1988</a>)</span>.</p>
          <p>As shown in <a href="#fig-numeric-distributions" class="quarto-xref">Figure&nbsp;1</a>, the numeric
            features of the dataset have varying distributions, which provide an overview of the dataâ€™s spread and
            tendencies.</p>
          <div id="fig-numeric-distributions" class="quarto-float quarto-figure quarto-figure-center anchored">
            <figure class="quarto-float quarto-float-fig figure">
              <div aria-describedby="fig-numeric-distributions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                <img src="results/figures/raw_feature_distributions.png" class="img-fluid figure-img"
                  style="width:70.0%">
              </div>
              <figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig"
                id="fig-numeric-distributions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                Figure&nbsp;1: Distribution of numeric features in the dataset.
              </figcaption>
            </figure>
          </div>
          <p>The target variable (<code>num</code>) is distributed as displayed in <a href="#fig-target-distribution"
              class="quarto-xref">Figure&nbsp;2</a>, where the majority of the samples are classified into the lower
            categories of heart disease.</p>
          <div id="fig-target-distribution" class="quarto-float quarto-figure quarto-figure-center anchored">
            <figure class="quarto-float quarto-float-fig figure">
              <div aria-describedby="fig-target-distribution-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                <img src="results/figures/target_variable_distribution.png" class="img-fluid figure-img"
                  style="width:75.0%">
              </div>
              <figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig"
                id="fig-target-distribution-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                Figure&nbsp;2: Distribution of the target variable (<code>num</code>).
              </figcaption>
            </figure>
          </div>
          <p>For categorical features, the grouped distributions based on the presence of heart disease are shown in <a
              href="#fig-categorical-distributions" class="quarto-xref">Figure&nbsp;3</a>. These stacked bar plots
            illustrate the relationship between categorical variables and the target variable.</p>
          <div id="fig-categorical-distributions" class="quarto-float quarto-figure quarto-figure-center anchored">
            <figure class="quarto-float quarto-float-fig figure">
              <div aria-describedby="fig-categorical-distributions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                <img src="results/figures/categorical_features_distribution.png" class="img-fluid figure-img"
                  style="width:70.0%">
              </div>
              <figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig"
                id="fig-categorical-distributions-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                Figure&nbsp;3: Distribution of categorical features grouped by heart disease (<code>num</code>).
              </figcaption>
            </figure>
          </div>
          <p>Boxplots for selected numeric features grouped by the target variable (num) are presented in <a
              href="#fig-boxplots-class" class="quarto-xref">Figure&nbsp;4</a>, offering insights into the spread and
            central tendencies of the selected features for each heart disease category.</p>
          <div id="fig-boxplots-class" class="quarto-float quarto-figure quarto-figure-center anchored">
            <figure class="quarto-float quarto-float-fig figure">
              <div aria-describedby="fig-boxplots-class-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                <img src="results/figures/raw_boxplots_by_class.png" class="img-fluid figure-img" style="width:75.0%">
              </div>
              <figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig"
                id="fig-boxplots-class-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                Figure&nbsp;4: Boxplots of selected numeric features grouped by heart disease (<code>num</code>).
              </figcaption>
            </figure>
          </div>
          <div style="background-color:#fce8e6; padding:15px; border-radius:5px; border: 2px solid #ff6666;">
            <h2 style="color:#ff6666;" class="anchored" data-anchor-id="data-set-summary">âœ… Visualization Section
              Completed</h2>
            <p style="font-size:14px; color:#cc0000;">
              This concludes the visualization section. The insights derived from these plots will be used to guide
              subsequent analyses
              and modeling steps. If additional visualizations are required, they can be added here.
            </p>
          </div>
        </section>
        <section id="discussion" class="level2" data-number="4.1">
          <h2 data-number="4.1" class="anchored" data-anchor-id="discussion"><span
              class="header-section-number">4.1</span> Discussion</h2>
          <div class="cell" data-execution_count="3">
            <div id="tbl-baseline_cv_results_df" class="cell quarto-float quarto-figure quarto-figure-center anchored"
              data-execution_count="3">
              <figure class="quarto-float quarto-float-tbl figure">
                <figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl"
                  id="tbl-baseline_cv_results_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  Table&nbsp;1: Baseline Model CV Results
                </figcaption>
                <div aria-describedby="tbl-baseline_cv_results_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  <div class="cell-output cell-output-display cell-output-markdown" data-execution_count="3">
                    <table class="do-not-create-environment cell caption-top table table-sm table-striped small">
                      <colgroup>
                        <col style="width: 21%">
                        <col style="width: 19%">
                        <col style="width: 19%">
                        <col style="width: 19%">
                        <col style="width: 19%">
                      </colgroup>
                      <thead>
                        <tr class="header">
                          <th style="text-align: left;"></th>
                          <th style="text-align: left;">fit_time</th>
                          <th style="text-align: left;">score_time</th>
                          <th style="text-align: left;">test_score</th>
                          <th style="text-align: left;">train_score</th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr class="odd">
                          <td style="text-align: left;">Dummy</td>
                          <td style="text-align: left;">0.009 (+/- 0.003)</td>
                          <td style="text-align: left;">0.005 (+/- 0.003)</td>
                          <td style="text-align: left;">0.543 (+/- 0.007)</td>
                          <td style="text-align: left;">0.542 (+/- 0.002)</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: left;">Decision tree</td>
                          <td style="text-align: left;">0.011 (+/- 0.003)</td>
                          <td style="text-align: left;">0.003 (+/- 0.004)</td>
                          <td style="text-align: left;">0.713 (+/- 0.048)</td>
                          <td style="text-align: left;">1.000 (+/- 0.000)</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: left;">SVC</td>
                          <td style="text-align: left;">0.010 (+/- 0.003)</td>
                          <td style="text-align: left;">0.005 (+/- 0.002)</td>
                          <td style="text-align: left;">0.844 (+/- 0.043)</td>
                          <td style="text-align: left;">0.929 (+/- 0.017)</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: left;">Logistic Regression</td>
                          <td style="text-align: left;">0.012 (+/- 0.001)</td>
                          <td style="text-align: left;">0.005 (+/- 0.002)</td>
                          <td style="text-align: left;">0.849 (+/- 0.027)</td>
                          <td style="text-align: left;">0.889 (+/- 0.015)</td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </figure>
            </div>
          </div>
          <div class="cell" data-execution_count="4">
            <div id="tbl-best_model_cv_results_df" class="cell quarto-float quarto-figure quarto-figure-center anchored"
              data-execution_count="4">
              <figure class="quarto-float quarto-float-tbl figure">
                <figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl"
                  id="tbl-best_model_cv_results_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  Table&nbsp;2: Best Model CV Results
                </figcaption>
                <div aria-describedby="tbl-best_model_cv_results_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  <div class="cell-output cell-output-display cell-output-markdown" data-execution_count="4">
                    <table class="do-not-create-environment cell caption-top table table-sm table-striped small">
                      <colgroup>
                        <col style="width: 21%">
                        <col style="width: 19%">
                        <col style="width: 19%">
                        <col style="width: 19%">
                        <col style="width: 19%">
                      </colgroup>
                      <thead>
                        <tr class="header">
                          <th style="text-align: left;"></th>
                          <th style="text-align: left;">fit_time</th>
                          <th style="text-align: left;">score_time</th>
                          <th style="text-align: left;">test_score</th>
                          <th style="text-align: left;">train_score</th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr class="odd">
                          <td style="text-align: left;">Decision Tree</td>
                          <td style="text-align: left;">0.012 (+/- 0.001)</td>
                          <td style="text-align: left;">0.004 (+/- 0.003)</td>
                          <td style="text-align: left;">0.778 (+/- 0.037)</td>
                          <td style="text-align: left;">0.960 (+/- 0.014)</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: left;">SVC</td>
                          <td style="text-align: left;">0.008 (+/- 0.003)</td>
                          <td style="text-align: left;">0.005 (+/- 0.003)</td>
                          <td style="text-align: left;">0.849 (+/- 0.040)</td>
                          <td style="text-align: left;">0.875 (+/- 0.007)</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: left;">Logistic Regression</td>
                          <td style="text-align: left;">0.012 (+/- 0.001)</td>
                          <td style="text-align: left;">0.006 (+/- 0.001)</td>
                          <td style="text-align: left;">0.858 (+/- 0.053)</td>
                          <td style="text-align: left;">0.881 (+/- 0.009)</td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </figure>
            </div>
          </div>
          <p>After doing checking on the baseline (<a href="#tbl-baseline_cv_results_df"
              class="quarto-xref">Table&nbsp;1</a>) we proceed to do the cross-validation. In the final cross-validation
            results (<a href="#tbl-best_model_cv_results_df" class="quarto-xref">Table&nbsp;2</a>), both the best SVC
            and best Logistic Regression achieve excellent test scores. The small gap between their training and test
            scores suggests that both models generalize well, with minimal overfitting. However, Logistic Regression has
            a smaller gap between training and test scores (0.023) compared to SVC (0.026), suggesting that it might
            generalize slightly better than SVC.</p>
          <div class="cell" data-execution_count="6">
            <div id="tbl-test_scores_df" class="cell quarto-float quarto-figure quarto-figure-center anchored"
              data-execution_count="6">
              <figure class="quarto-float quarto-float-tbl figure">
                <figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl"
                  id="tbl-test_scores_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  Table&nbsp;3: Test Score Results
                </figcaption>
                <div aria-describedby="tbl-test_scores_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  <div class="cell-output cell-output-display cell-output-markdown" data-execution_count="6">
                    <table class="do-not-create-environment cell caption-top table table-sm table-striped small">
                      <thead>
                        <tr class="header">
                          <th style="text-align: right;"></th>
                          <th style="text-align: left;">Model</th>
                          <th style="text-align: right;">Accuracy</th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr class="odd">
                          <td style="text-align: right;">0</td>
                          <td style="text-align: left;">SVC fitted</td>
                          <td style="text-align: right;">0.82</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">1</td>
                          <td style="text-align: left;">Logistic Regression fitted</td>
                          <td style="text-align: right;">0.84</td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </figure>
            </div>
          </div>
          <p>This is further confirmed by the test scores <a href="#tbl-test_scores_df"
              class="quarto-xref">Table&nbsp;3</a>, which show that Logistic Regression slightly outperforms SVC on
            unseen data.</p>
          <div class="cell" data-execution_count="7">
            <div id="tbl-coefficient_df" class="cell quarto-float quarto-figure quarto-figure-center anchored"
              data-execution_count="7">
              <figure class="quarto-float quarto-float-tbl figure">
                <figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl"
                  id="tbl-coefficient_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  Table&nbsp;4: Detailed Coefficients for Logistic Regression
                </figcaption>
                <div aria-describedby="tbl-coefficient_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  <div class="cell-output cell-output-display cell-output-markdown" data-execution_count="7">
                    <table class="do-not-create-environment cell caption-top table table-sm table-striped small">
                      <thead>
                        <tr class="header">
                          <th style="text-align: right;"></th>
                          <th style="text-align: left;">Feature</th>
                          <th style="text-align: right;">Coefficient</th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr class="odd">
                          <td style="text-align: right;">0</td>
                          <td style="text-align: left;">pipeline-1__ca</td>
                          <td style="text-align: right;">0.674653</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">1</td>
                          <td style="text-align: left;">onehotencoder__cp_4.0</td>
                          <td style="text-align: right;">0.614785</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">2</td>
                          <td style="text-align: left;">pipeline-2__thal_7.0</td>
                          <td style="text-align: right;">0.450091</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">3</td>
                          <td style="text-align: left;">standardscaler__oldpeak</td>
                          <td style="text-align: right;">0.400388</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">4</td>
                          <td style="text-align: left;">onehotencoder__exang_1.0</td>
                          <td style="text-align: right;">0.323725</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">5</td>
                          <td style="text-align: left;">onehotencoder__sex_1.0</td>
                          <td style="text-align: right;">0.297677</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">6</td>
                          <td style="text-align: left;">onehotencoder__slope_2.0</td>
                          <td style="text-align: right;">0.275565</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">7</td>
                          <td style="text-align: left;">standardscaler__trestbps</td>
                          <td style="text-align: right;">0.253341</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">8</td>
                          <td style="text-align: left;">onehotencoder__restecg_2.0</td>
                          <td style="text-align: right;">0.168724</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">9</td>
                          <td style="text-align: left;">onehotencoder__fbs_0.0</td>
                          <td style="text-align: right;">0.161019</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">10</td>
                          <td style="text-align: left;">standardscaler__age</td>
                          <td style="text-align: right;">0.0274616</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">11</td>
                          <td style="text-align: left;">standardscaler__chol</td>
                          <td style="text-align: right;">0.0143758</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">12</td>
                          <td style="text-align: left;">pipeline-2__thal_6.0</td>
                          <td style="text-align: right;">0.00637277</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">13</td>
                          <td style="text-align: left;">onehotencoder__restecg_1.0</td>
                          <td style="text-align: right;">-0.00791221</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">14</td>
                          <td style="text-align: left;">onehotencoder__cp_2.0</td>
                          <td style="text-align: right;">-0.055756</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">15</td>
                          <td style="text-align: left;">onehotencoder__slope_3.0</td>
                          <td style="text-align: right;">-0.124035</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">16</td>
                          <td style="text-align: left;">onehotencoder__slope_1.0</td>
                          <td style="text-align: right;">-0.15156</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">17</td>
                          <td style="text-align: left;">onehotencoder__restecg_0.0</td>
                          <td style="text-align: right;">-0.160842</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">18</td>
                          <td style="text-align: left;">onehotencoder__fbs_1.0</td>
                          <td style="text-align: right;">-0.161049</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">19</td>
                          <td style="text-align: left;">onehotencoder__cp_3.0</td>
                          <td style="text-align: right;">-0.245399</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">20</td>
                          <td style="text-align: left;">onehotencoder__sex_0.0</td>
                          <td style="text-align: right;">-0.297707</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">21</td>
                          <td style="text-align: left;">onehotencoder__cp_1.0</td>
                          <td style="text-align: right;">-0.31366</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">22</td>
                          <td style="text-align: left;">onehotencoder__exang_0.0</td>
                          <td style="text-align: right;">-0.323755</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">23</td>
                          <td style="text-align: left;">standardscaler__thalach</td>
                          <td style="text-align: right;">-0.406858</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">24</td>
                          <td style="text-align: left;">pipeline-2__thal_3.0</td>
                          <td style="text-align: right;">-0.456493</td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </figure>
            </div>
          </div>
          <p>To better understand the relationship between each feature and heart disease presence, we examine the
            coefficients obtained from the logistic regression model (<a href="#tbl-coefficient_df"
              class="quarto-xref">Table&nbsp;4</a>). Each coefficient indicates how the corresponding feature influences
            the likelihood of heart disease. Positive coefficients suggest that as the feature increases, the likelihood
            of having heart disease increases as well, while negative coefficients suggest the opposite.</p>
          <div id="fig-logreg-coefficients" class="quarto-float quarto-figure quarto-figure-center anchored">
            <figure class="quarto-float quarto-float-fig figure">
              <div aria-describedby="fig-logreg-coefficients-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                <img src="results/figures/log_reg_feature_coefficients.png" class="img-fluid figure-img"
                  style="width:85.0%">
              </div>
              <figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig"
                id="fig-logreg-coefficients-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                Figure&nbsp;5: Coefficient Table for Logistic Regression
              </figcaption>
            </figure>
          </div>
          <p>In <a href="#fig-logreg-coefficients" class="quarto-xref">Figure&nbsp;5</a>, we can see that features like
            <code>ca</code>, <code>oldpeak</code>, and <code>trestbps</code> have relatively high positive coefficients,
            meaning they strongly influence the prediction of heart disease. This makes sense, as research shows that
            high blood pressure is one of the most important causes of heart disease <span class="citation"
              data-cites="Fuchs2020">(<a href="#ref-Fuchs2020" role="doc-biblioref">Fuchs and Whelton 2020</a>)</span>.
            For oldpeak specifically, research shows that ST depression during exercise is linked to higher risk of
            heart disease<span class="citation" data-cites="Carlen2019">(<a href="#ref-Carlen2019"
                role="doc-biblioref">Carlen et al. 2019</a>)</span>. In contrast, features like <code>thalach</code>
            have large negative coefficients, suggesting they are linked to a lower likelihood of heart disease.
            Features like <code>age</code> and <code>chol</code>, however, show little impact, as their coefficients are
            close to zero.
          </p>
          <p>Interestingly, females (<code>sex</code> = 0) are more likely to be free of heart disease, as indicated by
            the large negative coefficient for <code>onehotencoder__sex_0.0</code>. In contrast, males (<code>sex</code>
            = 1) are more likely to have heart disease, as reflected by the high positive coefficient for
            <code>onehotencoder__sex_1.0</code>. This is supported by <span class="citation"
              data-cites="Regitz-Zagrosek2023">Regitz-Zagrosek and Gebhard (<a href="#ref-Regitz-Zagrosek2023"
                role="doc-biblioref">2023</a>)</span>, which highlights how biological sex differences, such as
            premenopausal women having a relative protection from coronary artery disease.
          </p>
          <p>However, there are some limitations of this study. First of all, as categorical features were split into
            multiple binary columns, interpreting the coefficients for these encoded variables can be tricky. It can be
            difficult to directly correlate the coefficients with the original feature, and whether this approach is
            reasonable should also be questioned.</p>
          <p>Additionally, while the modelâ€™s coefficients offer useful insights, they should be taken with caution.
            Further exploration into feature relationships and more advanced modeling techniques might be required to
            better understand the complexities of predicting heart disease.</p>
          <div class="cell" data-execution_count="8">
            <div id="tbl-misclassified_examples_df"
              class="cell quarto-float quarto-figure quarto-figure-center anchored" data-execution_count="8">
              <figure class="quarto-float quarto-float-tbl figure">
                <figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl"
                  id="tbl-misclassified_examples_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  Table&nbsp;5: Misclassified Examples
                </figcaption>
                <div aria-describedby="tbl-misclassified_examples_df-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
                  <div class="cell-output cell-output-display cell-output-markdown" data-execution_count="8">
                    <table class="do-not-create-environment cell caption-top table table-sm table-striped small">
                      <colgroup>
                        <col style="width: 2%">
                        <col style="width: 4%">
                        <col style="width: 4%">
                        <col style="width: 4%">
                        <col style="width: 8%">
                        <col style="width: 5%">
                        <col style="width: 4%">
                        <col style="width: 7%">
                        <col style="width: 7%">
                        <col style="width: 6%">
                        <col style="width: 7%">
                        <col style="width: 6%">
                        <col style="width: 4%">
                        <col style="width: 5%">
                        <col style="width: 9%">
                        <col style="width: 12%">
                      </colgroup>
                      <thead>
                        <tr class="header">
                          <th style="text-align: right;"></th>
                          <th style="text-align: right;">age</th>
                          <th style="text-align: right;">sex</th>
                          <th style="text-align: right;">cp</th>
                          <th style="text-align: right;">trestbps</th>
                          <th style="text-align: right;">chol</th>
                          <th style="text-align: right;">fbs</th>
                          <th style="text-align: right;">restecg</th>
                          <th style="text-align: right;">thalach</th>
                          <th style="text-align: right;">exang</th>
                          <th style="text-align: right;">oldpeak</th>
                          <th style="text-align: right;">slope</th>
                          <th style="text-align: right;">ca</th>
                          <th style="text-align: right;">thal</th>
                          <th style="text-align: right;">True Label</th>
                          <th style="text-align: right;">Predicted Label</th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr class="odd">
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">64</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">4</td>
                          <td style="text-align: right;">130</td>
                          <td style="text-align: right;">303</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">122</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">2</td>
                          <td style="text-align: right;">2</td>
                          <td style="text-align: right;">2</td>
                          <td style="text-align: right;">3</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">1</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">48</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">2</td>
                          <td style="text-align: right;">110</td>
                          <td style="text-align: right;">229</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">168</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">3</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">7</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">0</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">2</td>
                          <td style="text-align: right;">59</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">160</td>
                          <td style="text-align: right;">273</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">2</td>
                          <td style="text-align: right;">125</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">3</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">0</td>
                        </tr>
                        <tr class="even">
                          <td style="text-align: right;">3</td>
                          <td style="text-align: right;">58</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">4</td>
                          <td style="text-align: right;">100</td>
                          <td style="text-align: right;">234</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">156</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">0.1</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">7</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">0</td>
                        </tr>
                        <tr class="odd">
                          <td style="text-align: right;">4</td>
                          <td style="text-align: right;">64</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">3</td>
                          <td style="text-align: right;">140</td>
                          <td style="text-align: right;">335</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">158</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">0</td>
                          <td style="text-align: right;">3</td>
                          <td style="text-align: right;">1</td>
                          <td style="text-align: right;">0</td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </figure>
            </div>
          </div>
          <p>From <a href="#tbl-misclassified_examples_df" class="quarto-xref">Table&nbsp;5</a>, we can see that false
            Positives (e.g., index 0): Predicted as 1.0 (positive for heart disease), but true label is 0.0. This
            individual has a high cholesterol level (<code>chol</code> = 303.0), moderate <code>oldpeak</code>
            (<code>oldpeak</code> = 2.0), and significant <code>ca</code> = 2.0, which might make the model lean toward
            predicting heart disease incorrectly.</p>
          <p>False Negatives (e.g., indices 1, 2, 3, 4): Predicted as <code>0</code> (no heart disease), but true label
            is <code>1</code>. Many of these cases involve features like high <code>thalach</code> (e.g., 168.0, 158.0)
            and <code>slope</code> = 3.0 or 1.0, which the model might not weigh heavily enough.</p>
          <p>Overall the Logistic Regression model performs well and could be useful as a first-pass screening tool in a
            clinical setting, but there are ways we can make it even better. First, we can take a closer look at the
            misclassified examples and compare them to correctly classified ones. This could help us identify features
            or patterns the model struggles with and guide us in improving the features or adding new ones that capture
            important relationships.</p>
          <p>Next, we could test other classifiers to see if they perform better. For example, Random Forests are good
            at handling feature interactions automatically, which could help improve accuracy.</p>
          <p>Finally, instead of just giving a prediction, the model could provide a probability for each class. This
            would help clinicians understand how confident the model is in its predictions. For low-confidence cases,
            additional tests or evaluations could be done to avoid mistakes.</p>
          <p>These changes could make the model even more accurate and useful in practice.</p>
        </section>
      </section>
      <section id="references" class="level1" data-number="5">




      </section>

      <div id="quarto-appendix" class="default">
        <section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography">
          <h2 class="anchored quarto-appendix-heading">5 References</h2>
          <div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
            <div id="ref-Attia2023" class="csl-entry" role="listitem">
              Attia, Peter. 2023. <em>Peter on the Four Horsemen of Chronic Disease</em>. PeterAttiaMD.com. <a
                href="https://peterattiamd.com/peter-on-the-four-horsemen-of-chronic-disease/">https://peterattiamd.com/peter-on-the-four-horsemen-of-chronic-disease/</a>.
            </div>
            <div id="ref-Bui2024" class="csl-entry" role="listitem">
              Bui, Trang. 2024. <span>â€œCardiovascular Disease Is Rising Again After Years of Improvement.â€</span>
              <em>Stat News</em>. <a
                href="https://www.statnews.com/2024/10/15/cardiovascular-disease-rising-experts-on-causes/">https://www.statnews.com/2024/10/15/cardiovascular-disease-rising-experts-on-causes/</a>.
            </div>
            <div id="ref-Carlen2019" class="csl-entry" role="listitem">
              Carlen, A., M. Gustafsson, Ã…. Aneq, and E. Nylander. 2019. <span>â€œExercise-Induced ST Depression in an
                Asymptomatic Population Without Coronary Artery Disease.â€</span> <em>Scandinavian Cardiovascular
                Journal</em> 53 (4): 206â€“12. <a
                href="https://doi.org/10.1080/14017431.2019.1626021">https://doi.org/10.1080/14017431.2019.1626021</a>.
            </div>
            <div id="ref-CDC2022" class="csl-entry" role="listitem">
              Centers for Disease Control and Prevention (CDC). 2022. <span>â€œLeading Causes of Death.â€</span>
              <em>National Center for Health Statistics</em>. <a
                href="https://www.cdc.gov/nchs/fastats/leading-causes-of-death.htm">https://www.cdc.gov/nchs/fastats/leading-causes-of-death.htm</a>.
            </div>
            <div id="ref-Detrano1988" class="csl-entry" role="listitem">
              Detrano, R., A. JÃ¡nosi, W. Steinbrunn, M. Pfisterer, J. Schmid, S. Sandhu, K. Guppy, S. Lee, and V.
              Froelicher. 1988. <span>â€œHeart Disease Data Set.â€</span> In <em>Proceedings of Machine Learning and
                Medical Applications</em>. UC Irvine Machine Learning Repository. <a
                href="https://archive.ics.uci.edu/dataset/45/heart+disease">https://archive.ics.uci.edu/dataset/45/heart+disease</a>.
            </div>
            <div id="ref-Fuchs2020" class="csl-entry" role="listitem">
              Fuchs, F. D., and P. K. Whelton. 2020. <span>â€œHigh Blood Pressure and Cardiovascular Disease.â€</span>
              <em>Hypertension</em> 75 (2): 285â€“92. <a
                href="https://doi.org/10.1161/HYPERTENSIONAHA.119.14240">https://doi.org/10.1161/HYPERTENSIONAHA.119.14240</a>.
            </div>
            <div id="ref-Regitz-Zagrosek2023" class="csl-entry" role="listitem">
              Regitz-Zagrosek, V., and C. Gebhard. 2023. <span>â€œGender Medicine: Effects of Sex and Gender on
                Cardiovascular Disease Manifestation and Outcomes.â€</span> <em>Nature Reviews Cardiology</em> 20 (4):
              236â€“47. <a
                href="https://doi.org/10.1038/s41569-022-00797-4">https://doi.org/10.1038/s41569-022-00797-4</a>.
            </div>
          </div>
        </section>
      </div>
    </main> <!-- /main -->
    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const toggleBodyColorMode = (bsSheetEl) => {
          const mode = bsSheetEl.getAttribute("data-mode");
          const bodyEl = window.document.querySelector("body");
          if (mode === "dark") {
            bodyEl.classList.add("quarto-dark");
            bodyEl.classList.remove("quarto-light");
          } else {
            bodyEl.classList.add("quarto-light");
            bodyEl.classList.remove("quarto-dark");
          }
        }
        const toggleBodyColorPrimary = () => {
          const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
          if (bsSheetEl) {
            toggleBodyColorMode(bsSheetEl);
          }
        }
        toggleBodyColorPrimary();
        const icon = "î§‹";
        const anchorJS = new window.AnchorJS();
        anchorJS.options = {
          placement: 'right',
          icon: icon
        };
        anchorJS.add('.anchored');
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function (e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button,
              {
                trigger: "manual",
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]
              });
            tooltip.show();
          }
          setTimeout(function () {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function (trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          // For code content inside modals, clipBoardJS needs to be initialized with a container option
          // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
        var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i = 0; i < links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function (el) {
              return el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'quarto',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
          window.tippy(el, config);
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i = 0; i < noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function () {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch { }
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const xrefs = window.document.querySelectorAll('a.quarto-xref');
        const processXRef = (id, note) => {
          // Strip column container classes
          const stripColumnClz = (el) => {
            el.classList.remove("page-full", "page-columns");
            if (el.children) {
              for (const child of el.children) {
                stripColumnClz(child);
              }
            }
          }
          stripColumnClz(note)
          if (id === null || id.startsWith('sec-')) {
            // Special case sections, only their first couple elements
            const container = document.createElement("div");
            if (note.children && note.children.length > 2) {
              container.appendChild(note.children[0].cloneNode(true));
              for (let i = 1; i < note.children.length; i++) {
                const child = note.children[i];
                if (child.tagName === "P" && child.innerText === "") {
                  continue;
                } else {
                  container.appendChild(child.cloneNode(true));
                  break;
                }
              }
              if (window.Quarto?.typesetMath) {
                window.Quarto.typesetMath(container);
              }
              return container.innerHTML
            } else {
              if (window.Quarto?.typesetMath) {
                window.Quarto.typesetMath(note);
              }
              return note.innerHTML;
            }
          } else {
            // Remove any anchor links if they are present
            const anchorLink = note.querySelector('a.anchorjs-link');
            if (anchorLink) {
              anchorLink.remove();
            }
            if (window.Quarto?.typesetMath) {
              window.Quarto.typesetMath(note);
            }
            // TODO in 1.5, we should make sure this works without a callout special case
            if (note.classList.contains("callout")) {
              return note.outerHTML;
            } else {
              return note.innerHTML;
            }
          }
        }
        for (var i = 0; i < xrefs.length; i++) {
          const xref = xrefs[i];
          tippyHover(xref, undefined, function (instance) {
            instance.disable();
            let url = xref.getAttribute('href');
            let hash = undefined;
            if (url.startsWith('#')) {
              hash = url;
            } else {
              try { hash = new URL(url).hash; } catch { }
            }
            if (hash) {
              const id = hash.replace(/^#\/?/, "");
              const note = window.document.getElementById(id);
              if (note !== null) {
                try {
                  const html = processXRef(id, note.cloneNode(true));
                  instance.setContent(html);
                } finally {
                  instance.enable();
                  instance.show();
                }
              } else {
                // See if we can fetch this
                fetch(url.split('#')[0])
                  .then(res => res.text())
                  .then(html => {
                    const parser = new DOMParser();
                    const htmlDoc = parser.parseFromString(html, "text/html");
                    const note = htmlDoc.getElementById(id);
                    if (note !== null) {
                      const html = processXRef(id, note);
                      instance.setContent(html);
                    }
                  }).finally(() => {
                    instance.enable();
                    instance.show();
                  });
              }
            } else {
              // See if we can fetch a full url (with no hash to target)
              // This is a special case and we should probably do some content thinning / targeting
              fetch(url)
                .then(res => res.text())
                .then(html => {
                  const parser = new DOMParser();
                  const htmlDoc = parser.parseFromString(html, "text/html");
                  const note = htmlDoc.querySelector('main.content');
                  if (note !== null) {
                    // This should only happen for chapter cross references
                    // (since there is no id in the URL)
                    // remove the first header
                    if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                      note.children[0].remove();
                    }
                    const html = processXRef(null, note);
                    instance.setContent(html);
                  }
                }).finally(() => {
                  instance.enable();
                  instance.show();
                });
            }
          }, function (instance) {
          });
        }
        let selectedAnnoteEl;
        const selectorForAnnotation = (cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' + annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
        // Handle positioning of the toggle
        window.addEventListener(
          "resize",
          throttle(() => {
            elRect = undefined;
            if (selectedAnnoteEl) {
              selectCodeLines(selectedAnnoteEl);
            }
          }, 10)
        );
        function throttle(fn, ms) {
          let throttle = false;
          let timer;
          return (...args) => {
            if (!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
            } else { // all the others get throttled
              if (timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
            }
          };
        }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i = 0; i < bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function () {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function (cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
    </script>
  </div> <!-- /content -->




</body>

</html>